elide-core/src/main/java/com/yahoo/elide/resources/JsonApiEndpoint.java

```diff
@@ -112,13 +113,29 @@ public Response patch(
      */
     @DELETE
     @Path("{path:.*}")
+    @Consumes("application/vnd.api+json")
     public Response delete(
         @PathParam("path") String path,
         @Context SecurityContext securityContext,
         String jsonApiDocument) {
         return build(elide.delete(path, jsonApiDocument, getUser.apply(securityContext)));
     }
 
+    /**
+     * Delete resource handler (expects no body and no content-type header).
+     *
+     * @param path request path
+     * @param securityContext security context
+     * @return response
+     */
+    @DELETE
+    @Path("{path:.*}")
+    public Response deleteWithoutBody(
```

Comment:
@clayreimann Fixing #321 isn't urgent because it can easily be fixed by subclasing JsonApiEndpoint in application and providing necessary DELETE method (actually it's what I've done in my app to continue work right now). So I don't insist on anything.

One thought is if we rename signatures in the way you propose - it would be good to ensure that deleteRelationship method can be used only on relationships URLs and delete method can be used only on resource URLs. Maybe some runtime checks or something. Without those checks new method names can be quite misleading because it will be still possible to delete resource using deleteRelationship method.